Resources/public/js/recommendationrestclient.js

```diff
@@ -59,12 +60,29 @@
             }
         };
 
-        var attributes = '';
         for (var i = 0; i < this.attributes.length; i++) {
             attributes = attributes + '&attribute=' + this.attributes[i];
         }
 
-        xmlhttp.open('GET', this.endpointUrl + this.scenario + '.json' + '?numrecs=' + this.limit + '&' + 'contextitems=' + this.contextItems + attributes + '&contenttype=' + this.contentType + '&outputtype=' + this.outputType + '&categorypath=' + encodeURIComponent(this.categoryPath) + '&lang=' + this.language, true);
+        requestQueryString.push(
+            this.endpointUrl,
+            this.scenario,
+            '.json?numrecs=',
+            this.limit,
+            '&contextitems=',
+            this.contextItems,
+            attributes,
+            '&contenttype=',
+            this.contentType,
+            '&outputtype=',
+            this.outputType,
+            '&categorypath=',
+            encodeURIComponent(this.categoryPath),
+            '&lang=',
+            this.language
+        );
```

Comment:
From my POV I have all the params and their values visible in one place, but instead of pushing new vaules to an array I would just create an array instance with values defined. Then we could avoid calling `push` method.